Model/BTCPay/BTCPayService.php

```diff
@@ -205,6 +205,13 @@ public function createInvoice(Order $order): array
 
         $ba = $order->getBillingAddress();
         $sa = $order->getShippingAddress();
+      
+        // if there is no shipping address, use the billing address
+        // this happens if all products in an order are virtual or downloadable
+        // and no shipping is required
+        if (!$sa) {
```

Comment:
Please use `$order->isVirtual()`

Reply:
i dont know why, but this doesnt work for me, it doesnt trigger, $sa is still null in my case and throws.

Reply:
That does not sound health. Can you debug? Might be something with your store then...
if all items in the order are virtual (i.e. not delivery needed), it should say the order itself is virtual and there won't be a shipping address presented at checkout.

My solution is Magento best practise and we should aim to do it this way.

Reply:
`Invalid method Magento\Sales\Model\Order\Interceptor
▶ 268                 \Magento\Framework\Profiler::stop('magento');                   |::isVirtual

Reply:
![screenshot-1639390127](https://user-images.githubusercontent.com/1743657/145792982-ccc966d3-5eed-4836-9e8f-de2e5a4b821c.jpg)

Reply:
somehow data in the order is_virtual is 1 but the function retursn false... im very confused, maybe the isVirtual() doesnt exist anymore?